feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/CameraControlsOverlay.kt

```diff
@@ -249,7 +251,18 @@ private fun CaptureButton(
             multipleEventsCutter.processEvent {
                 when (previewMode) {
                     is PreviewMode.StandardMode -> {
-                        onCaptureImage()
+                        val timeStamp = System.currentTimeMillis()
+                        onCaptureImageWithUri(
+                            context.contentResolver,
+                            Uri.fromFile(
+                                File(
+                                    Environment.getExternalStoragePublicDirectory(
```

Comment:
This method was deprecated in API 29. We should be using something similar to the core test app on API 29+: https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:camera/integration-tests/coretestapp/src/main/java/androidx/camera/integration/core/CameraXActivity.java;l=975;drc=92cfd0386ecb506581413d8ecee586c8f648da7c

Note: you can still do this by generating a URI with ContentResolver.resolve() and use the same `onCaptureImageWithUri()` method. You'll probably want to create a helper function for getting the URI based on API level checks.

Reply:
I changed the case in CameraXCameraUseCase.takePicture() where the provided uri is null to generate the same outputFileOptions as in the linked code. The result is the same: image is saved to the pictures directory.